### DISCUSSION TITLE
NextJS SSG + PWA + Vercel?

### DISCUSSION THREAD

**Discussion by watch-janick** [vercel/next.js#17748] [Help] [answered]
Hello,

Is it possible to combine SSG + PWA + Vercel?

On all the PWA example that I've checked they use a custom server which isn't compatible with Vercel.

Any issues using SSG with PWA?

If you could share your experience combining these features together that would be awesome!

Thanks :)

**Comment by jamesmosier** [comment]
Hi @watch-janick.

There are a couple community plugins that allow you to combine SSG + PWA without a custom server:

[next-pwa](https://github.com/shadowwalker/next-pwa#usage-without-custom-server-nextjs-9) and [next-offline](https://github.com/hanford/next-offline#now-20)

**Reply by watch-janick** [reply]
@jamesmosier Awesome, Thanks! I guess it works on Vercel too. I'll have to try that!

**Comment by leerob** [comment, marked as answer]
Hey everyone. We now have updated documentation on building PWAs with Next.js.

https://nextjs.org/docs/app/building-your-application/configuring/progressive-web-apps
